### PR TITLE
feat: add `noPrimaryKey` to prevent creating a default PK attribute

### LINKED ISSUE #14385: Add optional `noPrimaryKey` to prevent creating a default PK attribute
## Issue Creation Checklist

[x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

### Is your feature request related to a problem? Please describe.

Yes, use new option called `noPrimaryKey` similar to `timestamps` instead of manually remove it using `removeAttribute`.
The reference code to work with: https://github.com/sequelize/sequelize/blob/e59b3ff23d9c768994c502ee5be60279cb46a7e5/lib/model.js#L224 

### Describe the solution you'd like

A clear and concise description of what you want to happen. How can the requested feature be used to approach the problem it's supposed to solve?

```js
Model.init({}. { noPrimaryKey: true });
```

### Describe alternatives/workarounds you've considered

Add the` noPrimaryKey` option, if false and no primary key in the definitions, add the primary key `id` field automatically,

## Feature Request Checklist

- [x] Add the` noPrimaryKey` option, if false and no primary key in the definitions, add the primary key `id` field automatically,

### Is this feature dialect-specific?

- [x] No. This feature is relevant to Sequelize as a whole.

### Would you be willing to implement this feature by submitting a Pull Request?

- [x] Yes, I have the time and I know how to start.

### PR DESCRIPTION
**Pull request by fzn0x**
### Pull Request Checklist

_Please make sure to review and check all of these items:_

- [x] Have you added new tests to prevent regressions?
- [x] Does `yarn test` or `yarn test-DIALECT` pass with this change (including linting)?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

Closes https://github.com/sequelize/sequelize/issues/3142 , this is unlikely breaking changes, but further review might needed.
Closes https://github.com/sequelize/sequelize/issues/11830

After this being merged, new section about automatic primary key field creation will be added here https://sequelize.org/docs/v6/core-concepts/model-basics/

### Todos

- [x] Add the noPrimaryKey option, if false and no primary key in the definitions, add the primary key id field automatically.

### CODE REVIEW

File: src/model.js [hunk 1]
```diff
@@ -256,6 +256,10 @@ export class Model {
           _autoGenerated: true,
         },
       };
+
+      if (Object.keys(this.primaryKeys).length === 0) {
```
Comment (WikiRik) [inline src/model.js:260]: Interesting that this doesn't fail a test. Since in line 243 it mentioned that `this.primaryKeys` can't be used there. Is that outdated documentation or do we just not have proper tests for that?

Comment (fzn0x) [reply src/model.js:260]: > Interesting that this doesn't fail a test. Since in line 243 it mentioned that `this.primaryKeys` can't be used there. Is that outdated documentation or do we just not have proper tests for that?

I think it was outdated documentation, and it feels like the line was for default primary key when none is defined in attributes object, so I put it there and everything goes okay.

Comment (fzn0x) [reply src/model.js:260]: The comment was `we can't use here` and basically no errors now. So I think it's safe to delete this outdated documentation for now. 😃

**Comment by fzn0x** [comment]
I think it's the correct logic, for now, we shouldn't force developers to always mark `id` as the primary key since id e,g could be a data first-place for data scrapping before replicated, for example, if you store data from an API that has `id` response and you create auto field creation and unfortunately hows previous logic handles gonna throw an error.

**Comment by fzn0x** [comment]
Related to https://github.com/sequelize/sequelize/issues/3142

File: src/model.d.ts [hunk 2]
```diff
@@ -1703,6 +1703,11 @@ export interface ModelOptions<M extends Model = Model> {
    */
   omitNull?: boolean;
 
+  /**
+   * Adds primary key automatically to the model if false. Default false.
```
Comment (ephys) [inline src/model.d.ts:1707]: use `@default` for default values, TypeDoc will render it :)

```suggestion
   * Sequelize will automatically add a primary key called `id` if no
   * primary key has been added manually.
   *
   * Set to false to disable adding that primary key.
   *
   * @default false
```

File: test/unit/model/define.test.js [hunk 3]
```diff
@@ -30,20 +30,12 @@ describe(Support.getTestDialectTeaser('Model'), () => {
       expect(Model.rawAttributes).not.to.have.property('updated_at');
     });
 
-    it('should throw when id is added but not marked as PK', () => {
```
Comment (ephys) [inline test/unit/model/define.test.js:33]: this test should not have been removed. It's not incompatible with `noPrimaryKey`

File: src/model.js (hunk 1 above)
Comment (ephys) [reply src/model.js:254]: I can confirm that by the time `_addDefaultAttributes` is called, `this.primaryKeys` is always equal to an empty object. It's populated by `refreshAttributes` which is called after `_addDefaultAttributes`

These couple of lines do nothing in theory, `primaryKeys` will be overwritten as soon as this function ends

It also means that the comment
> Can't use this.primaryKeys here, since this function is called before PKs are identified

is right, we can't use `this.primaryKeys` in here

File: src/model.js [hunk 4]
```diff
@@ -240,13 +240,7 @@ export class Model {
...
 
     // Add id if no primary key was manually added to definition
-    // Can't use this.primaryKeys here, since this function is called before PKs are identified
-    if (!_.some(this.rawAttributes, 'primaryKey')) {
-      if ('id' in this.rawAttributes) {
-        // Something is fishy here!
-        throw new Error(`A column called 'id' was added to the attributes of '${this.tableName}' but not marked with 'primaryKey: true'`);
-      }
-
+    if (!_.some(this.rawAttributes, 'primaryKey') && !this.options.noPrimaryKey) {
```
Comment (ephys) [inline src/model.js:243]:
```suggestion
    if (!this.options.noPrimaryKey && !_.some(this.rawAttributes, 'primaryKey')) {
```

better performances even though it's unnoticeable because this code runs rarely

File: src/model.js [hunk 5]
```diff
@@ -240,13 +240,7 @@ export class Model {
     let head = {};
 
     // Add id if no primary key was manually added to definition
-    // Can't use this.primaryKeys here, since this function is called before PKs are identified
-    if (!_.some(this.rawAttributes, 'primaryKey')) {
-      if ('id' in this.rawAttributes) {
```
Comment (ephys) [inline src/model.js:245]: this code should not have been removed

**Review by ephys — CHANGES_REQUESTED** [review]
Because the logic around it has been changed, can you add the following tests as well:

- test that the default PK `id` is added if no PK has been defined manually and `noPrimaryKey` is not set
- test that the default PK `id` is not added if a PK has been defined manually
- test that `new Sequelize({ define: { noPrimaryKey: true } })` is supported

**Comment by ephys** [comment]
> I think it's the correct logic, for now, we shouldn't force developers to always mark id as the primary key since id e,g could be a data first-place for data scrapping before replicated, for example, if you store data from an API that has id response and you create auto field creation and unfortunately hows previous logic handles gonna throw an error.

Has this been an issue for anyone so far? If so we could throw only if `this.rawAttributes.id.primaryKey` is `undefined` (not false) and update the error message to

```js
throw new Error(`An attribute called 'id' was defined in model '${this.tableName}' but not marked as a primaryKey. This is likely to be an error, which can be fixed by setting its 'primaryKey' option to true. If this is intended, explicitly set its 'primaryKey' option to false`);
```

**Comment by fzn0x** [comment]
> > I think it's the correct logic, for now, we shouldn't force developers to always mark id as the primary key since id e,g could be a data first-place for data scrapping before replicated, for example, if you store data from an API that has id response and you create auto field creation and unfortunately hows previous logic handles gonna throw an error.
> 
> Has this been an issue for anyone so far? If so we could throw only if `this.rawAttributes.id.primaryKey` is `undefined` (not false) and update the error message to
> 
> ```js
> throw new Error(`An attribute called 'id' was defined in model '${this.tableName}' but not marked as a primaryKey. This is likely to be an error, which can be fixed by setting its 'primaryKey' option to true. If this is intended, explicitly set its 'primaryKey' option to false`);
> ```

Mmmh it's a no, it's just my personal preference 😄 that's why I need further review, thanks for both of you!

**Comment by fzn0x** [comment]
Should `this.primaryKeys` removed @ephys ? since it will called after `_addDefaultAttributes`?

**Comment by fzn0x** [comment]
> Because the logic around it has been changed, can you add the following tests as well:
> 
> * test that the default PK `id` is added if no PK has been defined manually and `noPrimaryKey` is not set
> * test that the default PK `id` is not added if a PK has been defined manually
> * test that `new Sequelize({ define: { noPrimaryKey: true } })` is supported

Added, thanks!

Can we continue to https://github.com/sequelize/sequelize/pull/14386#issuecomment-1099114666? Let's ship this fast and good!

File: package.json [hunk 6]
```diff
@@ -215,7 +215,7 @@
     "docs": "typedoc",
     "----------------------------------------- tests ---------------------------------------------------": "",
     "mocha": "mocha -r ./test/registerEsbuild",
-    "test-unit": "yarn mocha \"test/unit/**/*.test.[tj]s\"",
```
Comment (ephys) [inline package.json:218]: if you want to run a single test suite, you can use `describe.only`. eslint will mark it as an error to stop you from committing it :)

**Comment by fzn0x** [comment]
I didn't know that, thanks 😃

File: test/unit/model/define.test.js [hunk 7]
```diff
@@ -35,15 +35,60 @@ describe(Support.getTestDialectTeaser('Model'), () => {
...
+
+    it('should not add the default `id` field PK if PK has been defined manually', () => {
+      const Model = current.define('User', {
+        id: {
+          type: DataTypes.INTEGER,
+          primaryKey: true,
+        },
+      });
+
+      expect(Model.rawAttributes).to.have.property('id');
```
Comment (ephys) [inline test/unit/model/define.test.js:79]: for this test to work, the PK should be named something different than 'id', otherwise we can't test that 'id' hasn't been added

```suggestion
      const Model = current.define('User', {
        customId: {
          type: DataTypes.INTEGER,
          primaryKey: true,
        },
      });

      expect(Model.rawAttributes).not.to.have.property('id');
```

File: test/unit/model/define.test.js [hunk 8]
```diff
@@ -35,15 +35,60 @@ describe(Support.getTestDialectTeaser('Model'), () => {
...
           id: {
             type: DataTypes.INTEGER,
           },
         });
-      }).to.throw('A column called \'id\' was added to the attributes of \'bars\' but not marked with \'primaryKey: true\'');
+      }).to.throw('An attribute called \'id\' was defined in model \'bars\' but not marked as a primaryKey. This is likely to be an error, which can be fixed by setting its \'primaryKey\' option to true. If this is intended, explicitly set its \'primaryKey\' option to false');
+    });
+
+    it('should allow model definition without PK', () => {
+      const Model = current.define('User', {}, {
```
Comment (ephys) [inline test/unit/model/define.test.js:50]: nit: avoid naming Models 'Model' to disambiguate with the 'Model' class provided by Sequelize

```suggestion
      const User = current.define('User', {}, {
```

File: test/unit/model/define.test.js [hunk 9]
```diff
@@ -35,15 +35,60 @@ describe(Support.getTestDialectTeaser('Model'), () => {
...
         current.define('bar', {
           id: {
             type: DataTypes.INTEGER,
           },
         });
-      }).to.throw('A column called \'id\' was added to the attributes of \'bars\' but not marked with \'primaryKey: true\'');
+      }).to.throw('An attribute called \'id\' was defined in model \'bars\' but not marked as a primaryKey. This is likely to be an error, which can be fixed by setting its \'primaryKey\' option to true. If this is intended, explicitly set its \'primaryKey\' option to false');
+    });
+
+    it('should allow model definition without PK', () => {
```
Comment (ephys) [inline test/unit/model/define.test.js:49]:
```suggestion
    it('should not add the default PK when noPrimaryKey is set to true', () => {
```

File: test/unit/model/define.test.js [hunk 10]
```diff
@@ -35,15 +35,60 @@ describe(Support.getTestDialectTeaser('Model'), () => {
...
+      expect(Model.rawAttributes).not.to.have.property('id');
+    });
+
+    it('should allow model definition without PK with noPrimaryKey is set to true', () => {
+      const Model = current.define('User', {}, {
+        noPrimaryKey: true,
+      });
+
+      expect(Model.rawAttributes).not.to.have.property('id');
+    });
```
Comment (ephys) [inline test/unit/model/define.test.js:63]: this test is exactly the same as the previous one and can be removed

File: test/unit/model/define.test.js [hunk 11]
```diff
@@ -35,15 +35,60 @@ describe(Support.getTestDialectTeaser('Model'), () => {
...
+
+    it('should add the default `id` field PK if noPrimary is set', () => {
```
Comment (ephys) [inline test/unit/model/define.test.js:65]:
```suggestion
    it('should add the default `id` field PK if noPrimary is not set and no PK has been defined manually', () => {
```

File: test/unit/model/define.test.js [hunk 12]
```diff
@@ -35,15 +35,60 @@ describe(Support.getTestDialectTeaser('Model'), () => {
...
+        id: {
+          type: DataTypes.INTEGER,
+          primaryKey: true,
+        },
+      });
+
+      expect(Model.rawAttributes).to.have.property('id');
+    });
+
+    it('should support noPrimaryKey on Sequelize define attribute', () => {
```
Comment (ephys) [inline test/unit/model/define.test.js:82]:
```suggestion
    it('should support noPrimaryKey on Sequelize define option', () => {
```

File: src/model.js [hunk 13]
```diff
@@ -240,11 +240,9 @@ export class Model {
     let head = {};
 
     // Add id if no primary key was manually added to definition
-    // Can't use this.primaryKeys here, since this function is called before PKs are identified
-    if (!_.some(this.rawAttributes, 'primaryKey')) {
+    if (!this.options.noPrimaryKey && !_.some(this.rawAttributes, 'primaryKey')) {
       if ('id' in this.rawAttributes) {
-        // Something is fishy here!
-        throw new Error(`A column called 'id' was added to the attributes of '${this.tableName}' but not marked with 'primaryKey: true'`);
+        throw new Error(`An attribute called 'id' was defined in model '${this.tableName}' but not marked as a primaryKey. This is likely to be an error, which can be fixed by setting its 'primaryKey' option to true. If this is intended, explicitly set its 'primaryKey' option to false`);
```
Comment (ephys) [inline src/model.js:245]: can you add a test to check that the following does not throw:

```js
current.define('bar', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: false,
  },
});
```

**Review by ephys — CHANGES_REQUESTED** [review]

**Comment by fzn0x** [comment]
Is this [comment](https://github.com/sequelize/sequelize/pull/14386#discussion_r850528507) mean to test `{ primaryKey: false }` to does not throw or does throw?

**Comment by fzn0x** [comment]
> > Should this.primaryKeys removed @ephys ? since it will populated by refreshAttributes after _addDefaultAttributes called?
> 
> Updating it in that method can be removed yes

I put the lines after `this.rawAttributes` and `this.primaryKeys` is declared. :)

**Comment by ephys** [comment]
> Is this https://github.com/sequelize/sequelize/pull/14386#discussion_r850528507 mean to test { primaryKey: false } to does not throw or does throw?

Does not throw:

```js
// this should throw
current.define('bar', {
  id: {
    type: DataTypes.INTEGER,
  },
});

// this should NOT throw
current.define('bar', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
  },
});

// this should NOT throw
current.define('bar', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: false,
  },
});
```

**Comment by fzn0x** [comment]
Already put it inside `this.rawAttributes` iteration, does it make sense to put it there?

**Comment by ephys** [comment]
Let me know when you're ready for another review

**Comment by fzn0x** [comment]
> Let me know when you're ready for another review

Cool, GMT +7 13:00 PM I will active :)

**Comment by fzn0x** [comment]
We will do next review next week, happy weekend @ephys @WikiRik

**Comment by fzn0x** [comment]
@ephys Let's continue with the review :)

**Comment by fzn0x** [comment]
I'd remove the `type: refactor` since the code changes is still part of `type: feature` as well. Just to decrease labels with something much more relevant and on point.

File: test/unit/model/define.test.js [hunk 14]
```diff
@@ -13,37 +13,84 @@ const dialect = Support.getTestDialect();
...
+      }).to.not.throw('An attribute called \'id\' was defined in model \'bars\' but primaryKey is not set. This is likely to be an error, which can be fixed by setting its \'primaryKey\' option to true. If this is intended, explicitly set its \'primaryKey\' option to false');
+
+      expect(() => {
+        current.define('baz', {
+          id: {
+            type: DataTypes.INTEGER,
+            primaryKey: false,
+          },
+        });
+      }).to.not.throw('An attribute called \'id\' was defined in model \'baz\' but primaryKey is not set. This is likely to be an error, which can be fixed by setting its \'primaryKey\' option to true. If this is intended, explicitly set its \'primaryKey\' option to false');
```
Comment (ephys) [inline test/unit/model/define.test.js:56]: This is not a good way to check that the function will not throw because it could throw with a slightly different message and the test will pass.

Move this to its own test and simply check that the field 'id' has been added to rawAttributes and is **not** a primary key.

File: test/unit/model/define.test.js [hunk 15]
```diff
@@ -13,37 +13,84 @@ const dialect = Support.getTestDialect();
...
 
       expect(() => {
         current.define('bar', {
           id: {
             type: DataTypes.INTEGER,
+            primaryKey: true,
           },
         });
-      }).to.throw('A column called \'id\' was added to the attributes of \'bars\' but not marked with \'primaryKey: true\'');
+      }).to.not.throw('An attribute called \'id\' was defined in model \'bars\' but primaryKey is not set. This is likely to be an error, which can be fixed by setting its \'primaryKey\' option to true. If this is intended, explicitly set its \'primaryKey\' option to false');
```
Comment (ephys) [inline test/unit/model/define.test.js:47]: This is not a good way to check that the function will not throw because it could throw with a slightly different message and the test will pass.

Move this to its own test and simply check that the field 'id' has been added to rawAttributes and is a primary key.

**Review by ephys — CHANGES_REQUESTED** [review]

**Comment by fzn0x** [comment]
I think _some didn't worked.

**Comment by fzn0x** [comment]
Done resolving @ephys, kindly check the latest changes :)

**Comment by fzn0x** [comment]
Having a problem when committing with eslint but eslint fix it for me, it looks like eslint fix the entire file, but nothing wrong with it since the changes already passed the lint build.

File: src/model.js [hunk 16]
```diff
@@ -240,11 +240,9 @@ export class Model {
     let head = {};
 
     // Add id if no primary key was manually added to definition
-    // Can't use this.primaryKeys here, since this function is called before PKs are identified
-    if (!_.some(this.rawAttributes, 'primaryKey')) {
-      if ('id' in this.rawAttributes) {
-        // Something is fishy here!
-        throw new Error(`A column called 'id' was added to the attributes of '${this.tableName}' but not marked with 'primaryKey: true'`);
+    if (!this.options.noPrimaryKey && !_.some(this.rawAttributes, 'primaryKey')) {
+      if ('id' in this.rawAttributes && _.get(this.rawAttributes, 'id.primaryKey') === undefined) {
```
Comment (ephys) [inline src/model.js:244]: Don't use `_.get`

```suggestion
      if ('id' in this.rawAttributes && this.rawAttributes.id.primaryKey === undefined) {
```

File: src/model.js [hunk 17]
```diff
@@ -1461,7 +1456,7 @@ export class Model {
     let indexes = await this.queryInterface.showIndex(tableName, options);
     indexes = this._indexes.filter(item1 => !indexes.some(item2 => item1.name === item2.name)).sort((index1, index2) => {
       if (this.sequelize.options.dialect === 'postgres') {
-      // move concurrent indexes to the bottom to avoid weird deadlocks
+        // move concurrent indexes to the bottom to avoid weird deadlocks
```
Comment (ephys) [inline src/model.js:1459]: avoid introducing unrelated style changes, this will cause unnecessary conflicts with the other PRs (also valable for the other style changes)

File: test/unit/model/define.test.js [hunk 18]
```diff
@@ -13,37 +13,90 @@ const dialect = Support.getTestDialect();
...
+          type: DataTypes.INTEGER,
+          primaryKey: true,
+        },
+      });
+
+      expect(Bar.rawAttributes).to.have.property('id');
+      expect(Bar.rawAttributes.id.primaryKey).to.equal(true);
+    });
+
+    it('should not a primary key field', () => {
```
Comment (ephys) [inline test/unit/model/define.test.js:53]:
```suggestion
    it('allows creating an "id" field explicitly marked as non primary key', () => {
```

File: test/unit/model/define.test.js [hunk 19]
```diff
@@ -13,37 +13,90 @@ const dialect = Support.getTestDialect();
...
+    });
 
-      expect(() => {
-        current.define('bar', {
-          id: {
-            type: DataTypes.INTEGER,
-          },
-        });
-      }).to.throw('A column called \'id\' was added to the attributes of \'bars\' but not marked with \'primaryKey: true\'');
+    it('should a primary key field', () => {
```
Comment (ephys) [inline test/unit/model/define.test.js:41]:
```suggestion
    it('allows creating an "id" field as the primary key', () => {
```

File: test/unit/model/define.test.js [hunk 20]
```diff
@@ -13,37 +13,90 @@ const dialect = Support.getTestDialect();
...
+    it('should not a primary key field', () => {
+      const Baz = current.define('baz', {
+        id: {
+          type: DataTypes.INTEGER,
+          primaryKey: false,
+        },
+      });
+
+      expect(Baz.rawAttributes).to.have.property('id');
+      expect(Baz.rawAttributes.id.primaryKey).to.equal(false);
```
Comment (ephys) [inline test/unit/model/define.test.js:62]: can you add one last expectation to ensure there are no primary keys in this model?

`Baz.primaryKeys` should be an empty object

**Review by ephys — CHANGES_REQUESTED** [review]
These are my last comments I think

**Comment by fzn0x** [comment]
It's all good now? let's merge it! :)

**Comment by fzn0x** [comment]
I was not saving it as deepEqual 😄

**Comment by fzn0x** [comment]
I think we are ready now :) @ephys @WikiRik

**Comment by fzn0x** [comment]
I think we are ready to merge now :) @ephys @WikiRik

**Comment by ephys** [comment]
@fzn0x you're missing this one :) https://github.com/sequelize/sequelize/pull/14386#discussion_r851663317

**Comment by fzn0x** [comment]
> @fzn0x you're missing this one :) https://github.com/sequelize/sequelize/pull/14386#discussion_r851663317

Oh haha, brb to my PC after shower

**Comment by fzn0x** [comment]
I think we are surely ready to merge now 😃 @ephys @WikiRik

if you guys still have review, go on :)

**Comment by fzn0x** [comment]
I think we are surely ready to merge now 😃 @ephys @WikiRik

if you guys still have review, go on :)

**Comment by fzn0x** [comment]
I think we are surely ready to merge now 😃 @ephys @WikiRik

**Comment by ephys** [comment]
But you haven't reverted the unnecessary code style changes yet

**Comment by fzn0x** [comment]
> But you haven't reverted the unnecessary code style changes yet

I think I have.. https://github.com/sequelize/sequelize/pull/14386/commits/e6b6cd896d7995352509185c972fcf67c15c8c86

**Comment by fzn0x** [comment]
> But you haven't reverted the unnecessary code style changes yet

Wait, let me reconstruct my commit

**Comment by fzn0x** [comment]
> But you haven't reverted the unnecessary code style changes yet

I will reconstruct my commit

**Comment by ephys** [comment]
Thanks :) looks good! I'll give it one last look in the morning and we can merge it

**Comment by fzn0x** [comment]
> Thanks :) looks good! I'll give it one last look in the morning and we can merge it

Stay healthy bro

**Review by ephys — APPROVED** [review]

**Comment by ephys** [comment]
Brilliant, thank you very much for implementing this :)

**Comment by damms005** [comment]
> After this being merged, new section about automatic primary key field creation will be added here https://sequelize.org/docs/v6/core-concepts/model-basics/

I think this is an important as this PR. I don't think this bit of the doc is fixed yet

**Comment by fzn0x** [comment]
> > > After this being merged, new section about automatic primary key field creation will be added here https://sequelize.org/docs/v6/core-concepts/model-basics/
> > 
> > I think this is an important as this PR. I don't think this bit of the doc is fixed yet
> 
> Indeed it's not. I suppose @fzn0x will open a PR for that soon on the [website](https://github.com/sequelize/website) repo. I do not expect this feature to land in v6 though

Yes, I was still working now, about a few hours will implement the documentation :)

**Comment by fzn0x** [comment]
PR open https://github.com/sequelize/website/pull/109
